packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,144 @@
+import { Construct } from 'constructs';
+import { CfnRealtimeLogConfig } from './cloudfront.generated';
+import * as iam from '../../aws-iam';
+import * as kinesis from '../../aws-kinesis';
+import { IResource, Lazy, Names, PhysicalName, Resource } from '../../core';
+
+/**
+ * @internal
+ */
+function singletonKinesisRole(scope: Construct): iam.IRole {
+  const id = 'RealtimeLogKinesisRole';
+  const existing = scope.node.tryFindChild(id) as iam.IRole;
+  if (existing) { return existing; }
+
+  const role = new iam.Role(scope, id, {
+    roleName: PhysicalName.GENERATE_IF_NEEDED,
+    assumedBy: new iam.ServicePrincipal('cloudfront.amazonaws.com'),
+  });
+
+  return role;
+}
+
+/**
+ * Represents the endpoints available for targetting within a realtime log config resource
+ */
+export abstract class Endpoint {
```

Comment:
At this point, this class probably needs to be moved into its own file. We can call it `endpoint.ts`. With that, we should also have unit tests for this.

Reply:
Moved - i moved some tests over too - to just get some coverage on the `render` part of this class